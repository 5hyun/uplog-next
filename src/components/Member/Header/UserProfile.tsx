import { FaUserCircle } from 'react-icons/fa';
import { AiFillStar, AiOutlinePoweroff } from 'react-icons/ai';
import { BiPencil } from 'react-icons/bi';
import { useNavigate } from 'react-router-dom';
import { profileOpen } from '@/recoil/User/atom.tsx';
import { useRecoilState } from 'recoil';

export default function UserProfile() {
  // TODO: 실제 userprofile 값으로 변경하기
  const userInfo = {
    userprofile: '',
    userName: '오채영',
    userNickname: 'OCI',
    userId: 'oco6029@naver.com',
  };

  const navigate = useNavigate();

  // userProfile click
  const [isProfileClick, setIsProfileClick] = useRecoilState(profileOpen);

  return (
    <section
      className={'border-base w-60 h-[15rem] absolute top-[3rem] right-[0rem] shadow-sign-up'}
    >
      <div
        className={
          'flex-row-center justify-between border-solid border-b-[1px] border-line w-full h-[5.5rem]'
        }
      >
        {/*유저 정보*/}
        <div className={'flex-row-center w-[4rem] h-full'}>
          {!userInfo.userprofile ? (
            <FaUserCircle className={'text-[2.8rem] fill-gray-dark ml-2.5'} />
          ) : (
            <img src={userInfo.userprofile} alt="userprofile" className={'w-[2.8rem] h-[2.8rem]'} />
          )}
        </div>
        <div className={'flex-col-center w-[10rem] h-full'}>
          <span
            className={'flex self-start text-lg font-bold'}
          >{`${userInfo.userNickname} (${userInfo.userName})`}</span>
          <span className={'flex self-start text-gray-light text-base font-semibold'}>
            {userInfo.userId}
          </span>
        </div>
      </div>
      {/*스크랩 페이지 이동*/}
      <div className={'flex-col-center w-full h-[9.5rem] cursor-pointer'}>
        <div
          className={'flex-row-center justify-between w-full h-1/3 hover:bg-hover'}
          onClick={() => {
            setIsProfileClick((prevState) => !prevState);
            navigate('/scrap');
          }}
        >
          <AiFillStar className={'text-4xl ml-4 fill-yellow'} />
          <span className={'w-full ml-3 text-[1.05rem] font-semibold cursor-pointer'}>스크랩</span>
        </div>
        {/*마이페이지 수정 이동*/}
        <div
          className={'flex-row-center justify-between w-full h-1/3 hover:bg-hover'}
          onClick={() => {
            setIsProfileClick(!isProfileClick);
            navigate('/mypage');
          }}
        >
          <BiPencil className={'text-4xl ml-4 fill-gray-light'} />
          <span className={'w-full ml-3 text-[1.05rem] font-semibold'}>프로필 수정</span>
        </div>
        {/*로그아웃*/}
        <div className={'flex-row-center justify-between w-full h-1/3 hover:bg-hover'}>
          <AiOutlinePoweroff className={'text-4xl ml-4 fill-gray-light'} />
          {/*TODO : 로그아웃 로직 추가하기*/}
          <span className={'w-full ml-3 text-[1.05rem] font-semibold'}>로그아웃</span>
        </div>
      </div>
    </section>
  );
}
