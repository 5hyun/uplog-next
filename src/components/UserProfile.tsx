import {FaUserCircle} from "react-icons/fa";

export default function UserProfile () {
    // TODO: 실제 userprofile 값으로 변경하기

    const userInfo = {
        userprofile : '',
        userName : '오채영',
        userNickname : 'OCI',
        userId : 'oco6029@naver.com'
    }

    return (
    <section className={'border-solid border-[1px] border-line bg-border w-52 h-[13.6rem] absolute top-[5.5rem] right-[3rem] shadow-sign-up'}>
        <div className={'flex-row-center justify-between border-solid border-b-[1px] border-line w-full h-[5.5rem]'}>
            <div className={'flex-row-center w-[4rem] h-full'}>
                {!userInfo.userprofile ? (
                    <FaUserCircle className={'text-[2.8rem] fill-gray-dark'} />
                ) : (
                    <img
                        src={userInfo.userprofile}
                        alt="userprofile"
                        className={'w-[2.8rem] h-[2.8rem]'}
                    />
                )}
            </div>
            <div className={'flex-col-center w-[9rem] h-full ml-1'}>
                <div className={'flex self-start text-lg'}>{`${userInfo.userNickname} (${userInfo.userName})`}</div>
                <div className={'flex self-start text-gray-light text-sm font-medium'}>{userInfo.userId}</div>
            </div>
        </div>
        <div className={'w-full h-[8.1rem] '}></div>
    </section>)
}