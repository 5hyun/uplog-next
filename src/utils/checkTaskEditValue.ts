import { TaskData } from '@/typings/task.ts';

export function checkTaskEditValue(editTask: TaskData): boolean {
  if (editTask.taskName === '') {
    return false;
  }
  if (editTask.targetMemberInfoDTO && editTask.targetMemberInfoDTO.id === 0) {
    return false;
  }
  if (editTask.menuId === 0) {
    return false;
  }
  if (editTask.teamId === 0) {
    return false;
  }
  if (editTask.startTime === '' || editTask.endTime === '') {
    return false;
  }

  return true;
}
