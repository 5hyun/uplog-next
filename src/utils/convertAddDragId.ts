import { StatusTaskData, Tasks } from '@/typings/task.ts';
import { useState } from 'react';

export function convertAddDragId(tasks: StatusTaskData): Tasks {
  const [convertedTasks, setConvertedTasks] = useState<Tasks>({
    PROGRESS_BEFORE: [],
    PROGRESS_IN: [],
    PROGRESS_COMPLETE: [],
  });

  const beforeStatusTasks = tasks.PROGRESS_BEFORE.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_BEFORE: beforeStatusTasks,
  });

  const inStatusTasks = tasks.PROGRESS_IN.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_IN: inStatusTasks,
  });

  const completeStatusTasks = tasks.PROGRESS_COMPLETE.map((task) => ({
    ...task,
    dragId: task.id.toString(), // dragId 값을 문자열로 변환하여 할당
  }));

  setConvertedTasks({
    ...convertedTasks,
    PROGRESS_COMPLETE: completeStatusTasks,
  });

  return convertedTasks;
}
