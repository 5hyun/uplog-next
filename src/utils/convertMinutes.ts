// ** 타이머 분:초 구하는 함수 */
export function convertMinutes(seconds: number): string {
  const minutes = Math.floor(seconds / 60);
  const remainingSeconds = seconds % 60;
  const formattedSeconds = remainingSeconds < 10 ? `0${remainingSeconds}` : `${remainingSeconds}`;

  if (minutes === 0) {
    return `00:${formattedSeconds}`;
  }

  return `${minutes}:${formattedSeconds}`;
}
