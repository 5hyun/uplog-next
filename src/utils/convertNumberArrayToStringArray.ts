export function convertNumberArrayToStringArray(numbers: number[]): string[] {
  // 배열의 각 요소를 문자열로 변환하여 새로운 배열을 생성합니다.
  const strings: string[] = numbers.map((number) => number.toString());
  return strings;
}
