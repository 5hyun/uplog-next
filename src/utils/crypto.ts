import CryptoJS from 'crypto-js';

/* Salt 비밀 키 */
// const salt = import.meta.env.VITE_SALT;
const salt = '1234';

/* 암호화 */
export const encrypt = (text: string) => {
  // 값이 없을 경우 빈 문자열 반환
  if (!text) return '';
  return CryptoJS.AES.encrypt(text, salt).toString();
};

/* 복호화 */
export const decrypt = (text: string) => {
  // 값이 없을 경우 빈 문자열 반환
  if (!text) return '';

  try {
    const bytes = CryptoJS.AES.decrypt(text, salt); // 복호화 시도
    return bytes.toString(CryptoJS.enc.Utf8);
  } catch (error) {
    return ''; // 에러 발생 시 빈 문자열 반환
  }
};
