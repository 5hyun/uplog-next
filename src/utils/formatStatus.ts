// ** 진행 전,중,후 상태로 변환하기 위한 함수 */

export function formatStatus(status: string): string {
  if (status === 'PROGRESS_BEFORE') {
    return '진행 전';
  } else if (status === 'PROGRESS_IN') {
    return '진행 중';
  }

  return '진행 후';
}
