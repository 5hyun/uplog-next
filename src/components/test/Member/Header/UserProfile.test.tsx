import { describe, expect, test } from '@jest/globals';
import { fireEvent, getByText, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { withAllContexts, withRouter } from '@/tests/utils';
import { Route, Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import UserProfile from '@/components/Member/Header/UserProfile';
import userEvent from '@testing-library/user-event';
import { RecoilRoot } from 'recoil';

describe('UserProfile', () => {
  const userInfo = {
    userprofile: '',
    userName: '오채영',
    userNickname: 'OCI',
    userId: 'oco6029@naver.com',
  };

  test('UserProfile Modal UI', () => {
    const component = renderer.create(
      withRouter(<Route path={'/'} element={<UserProfile />} />, '/workspace/product')
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('UserProfile Modal Open MenuInfo Correctly', () => {
    render(
      withRouter(
        <Route path={'/workspace/product'} element={<UserProfile />} />,
        '/workspace/product'
      )
    );

    expect(screen.getByText('스크랩')).toBeInTheDocument();
    expect(screen.getByText('프로필 수정')).toBeInTheDocument();
    expect(screen.getByText('로그아웃')).toBeInTheDocument();
  });
});
