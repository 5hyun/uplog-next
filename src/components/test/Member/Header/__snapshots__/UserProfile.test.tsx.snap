// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile UserProfile Modal UI 1`] = `null`;
